torch/utils/data/datapipes/gen_pyi.py

```diff
@@ -289,15 +316,14 @@ def main() -> None:
     )
 
     path = pathlib.Path(__file__).parent.resolve()
-    replacements = [
-        ("${IterDataPipeMethods}", iter_method_definitions, 4),
-        ("${MapDataPipeMethods}", map_method_definitions, 4),
-    ]
-    gen_from_template(
-        dir=str(path),
-        template_name="datapipe.pyi.in",
-        output_name="datapipe.pyi",
-        replacements=replacements,
+    fm = _FileManager(install_dir=path, template_dir=path, dry_run=False)
+    fm.write_with_template(
+        "datapipe.pyi",
+        "datapipe.pyi.in",
+        lambda: {
+            "IterDataPipeMethods": iter_method_definitions,
+            "MapDataPipeMethods": map_method_definitions,
+        },
     )
```

Comment:
> Sorry, what exactly does this do?

@ezyang This PR aims to reuse the functionality in `torchgen` such as template formatting.